web/pandas/pdeps/0018-nullable-object-dtype.md

```diff
@@ -0,0 +1,175 @@
+# PDEP-18: Nullable Object Dtype for Pandas
+
+- Created: 07 June 2025
+- Status: Draft
+- Discussion: [#32931](https://github.com/pandas-dev/pandas/issues/32931)
+- Author: [Simon Hawkins](https://github.com/simonjayhawkins)
+- Revision: 1
+
+## Abstract
+
+This proposal outlines the introduction of a nullable object
+dtype to the pandas library. The goal is to provide a
+dedicated dtype for handling arbitrary Python objects with
+consistent missing value semantics using `pd.NA`. Unlike the
+traditional `object` dtype which lacks robust missing data
+handling, this new nullable dtype will add clarity and
+consistency in representing missing or undefined values
+within object arrays.
+
+## Motivation
+
+Currently, the `object` dtype in pandas is a catch-all for
+heterogeneous Python objects, but it does not enforce any
+particular missing-value semantics. As pandas has evolved to
+include extension types (like `string[python]`, `Int64`, or
+`boolean`), there is a clear benefit in extending these
+improvements to the object datatype. A nullable object dtype
+would help:
+- **Consistency**: Enforce a uniform approach to managing
+missing values with `pd.NA` across all dtypes.
+- **Interoperability**: Enable cleaner and more predictable
+behavior when performing operations on data previously
+stored as generic objects.
+- **Clarity**: Help users distinguish between truly “object”
+data and data that is better represented by a nullable
+container supporting missing values.
+
+This proposal is driven by frequent community discussions
```

Comment:
are there links for any of these?  off the top of my head i dont recall any users asking for this in particular

Reply:
I plan to list the issues in the OP before officially opening the discussion period, but yes this does need references. Thanks for highlighting this.

Reply:
for the following part of the sentence

> ... and development efforts ...

i could potentially reference PDEP-16?